test/ttmlir/Dialect/TTNN/ccl/all_gather.mlir

```diff
@@ -0,0 +1,11 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+module attributes {} {
+  func.func @forward(%arg0: tensor<1x1x32x32xbf16>) -> tensor<1x1x32x128xbf16> {
+    // CHECK: %[[C:.*]] = "ttnn.empty"[[C:.*]]
+    %0 = tensor.empty() : tensor<1x1x32x128xbf16>
+    // CHECK: %[[C:.*]] = "ttnn.all_gather"[[C:.*]]
+    %1 = "ttir.all_gather"(%arg0, %0) <{dim = 3 : si32, num_links = 1 : si32, operand_constraints = [#any_device, #any_device]}> : (tensor<1x1x32x32xbf16>, tensor<1x1x32x128xbf16>) -> tensor<1x1x32x128xbf16>
+    return %1 : tensor<1x1x32x128xbf16>
+  }
+}
```

Comment:
possible to add equivalent silicon test for this op? if it's fully functional in runtime

Reply:
It isn't fully functional but I can add one silicon test and mark it UNSUPPORTED